source/common/grpc/google_async_client_impl.cc

```diff
@@ -176,11 +176,16 @@ void GoogleAsyncStreamImpl::initialize(bool /*buffer_body_for_retry*/) {
 void GoogleAsyncStreamImpl::notifyRemoteClose(Status::GrpcStatus grpc_status,
                                               Http::HeaderMapPtr trailing_metadata,
                                               const std::string& message) {
-  if (grpc_status > Status::GrpcStatus::MaximumValid) {
-    grpc_status = Status::GrpcStatus::Unknown;
+  if (grpc_status > Status::GrpcStatus::MaximumValid || grpc_status < 0) {
```

Comment:
Is the  `grpc_status < 0` check to make sure it's not already set to `InvalidCode`? If so can we just check that with equality? Can it be negative otherwise?

Reply:
But how can it be < 0? Can it be InvalidCode in this call path for the Google client? My preference would be to explicit check `|| grpc_status == Status::GrpcStatus::InvalidCode` and have an additional assert, but will defer to @htuch on this client.

Reply:
Yes it can be InvalidCode, which is the cause to issue #5127 

https://github.com/envoyproxy/envoy/blob/97fa8855d115aed81ebe423135b8f023f41f64a7/source/common/grpc/google_async_client_impl.cc#L329-L333

Reply:
Yeah, I think this is more robust; since the Google gRPC core code is outside our control, we don't know what it might throw at us, so best to be defensive here.